MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.mm

```diff
@@ -699,6 +740,24 @@ void mvkPopulateShaderConversionConfig(mvk::SPIRVToMSLConversionConfiguration& s
     }
 }
 
+bool MVKDescriptorSetLayoutBinding::needsBuffSizeAuxBuffer() {
+
+	if ( !_layout->isUsingMetalArgumentBuffers() ) { return false; }
+	if ( getDescriptorCount() == 0 ) { return false; }
+
+	switch (getDescriptorType()) {
+		case VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER:
+		case VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER_DYNAMIC:
+		case VK_DESCRIPTOR_TYPE_STORAGE_BUFFER:
+		case VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC:
+		case VK_DESCRIPTOR_TYPE_INLINE_UNIFORM_BLOCK_EXT:
```

Comment:
This is probably a good idea. For discrete bindings, we could likely also use `setVertexBytes:`, etc., to avoid a separate buffer.

Based on a quick triage, this might be a complex and invasive change. We'd have to track argument buffer offsets as well as current slot indexes, and Metal requires it to be declared as an array, and seems to consume a slot index for each element, which would be odd, if we treat it as a `char` array, as indexes would potentially jump by 4K at a time. And SPIRV-Cross will probably need some mods to accommodate recasting pointers, etc.

I do have plans for a couple of more PRs to come after this, including refactoring how we handle resource usage tracking (incl the new `MTLResidencySet`), so let's leave this out of the current PR, and investigate separately.